### PR TITLE
fix multiple save runs not working due to permissions

### LINKED ISSUE #11: Reciprocal modules copies read-only directories on save. Prevents multiple runs.
# Problem
The copy function used for `save` preserves all permissions, which means when files are copied from the module cache for Reciprocal/Restricted licenses all the files and directories are readonly as well. This means that multiple runs of `save` won't work because the directories containing 

# Example

```go
package example

import (
        _ "github.com/hashicorp/errwrap"
)
```

```sh
[2019-11-22 14:52:03 example ]$ go-licenses save . --save_path=third_party --force 
[2019-11-22 14:52:10 example ]$ go-licenses save . --save_path=third_party --force
Error: unlinkat ./third_party/github.com/hashicorp/errwrap/go.mod: permission denied
Usage:
  licenses save <package> [flags]

Flags:
      --force              Delete the destination directory if it already exists.
  -h, --help               help for save
      --save_path string   Directory into which files should be saved that are required by license terms

Global Flags:
      --alsologtostderr                  log to standard error as well as files
      --confidence_threshold float       Minimum confidence required in order to positively identify a license. (default 0.9)
      --log_backtrace_at traceLocation   when logging hits line file:N, emit a stack trace (default :0)
      --log_dir string                   If non-empty, write log files in this directory
      --logtostderr                      log to standard error instead of files
      --stderrthreshold severity         logs at or above this threshold go to stderr (default 2)
  -v, --v Level                          log level for V logs
      --vmodule moduleSpec               comma-separated list of pattern=N settings for file-filtered logging

F1122 14:52:12.566725  140815 main.go:43] unlinkat ./third_party/github.com/hashicorp/errwrap/go.mod: permission denied
[2019-11-22 14:52:12 example ]$ ls -lR third_party 
third_party:
total 4
drwxr-x--- 3 wlynch primarygroup 4096 Nov 22 14:52 github.com

third_party/github.com:
total 4
drwxr-x--- 3 wlynch primarygroup 4096 Nov 22 14:52 hashicorp

third_party/github.com/hashicorp:
total 4
dr-x------ 2 wlynch primarygroup 4096 Nov 22 14:52 errwrap

third_party/github.com/hashicorp/errwrap:
total 36
-r--r----- 1 wlynch primarygroup  4377 Nov 22 14:52 errwrap.go
-r--r----- 1 wlynch primarygroup  1292 Nov 22 14:52 errwrap_test.go
-r--r----- 1 wlynch primarygroup    36 Nov 22 14:52 go.mod
-r--r----- 1 wlynch primarygroup 15977 Nov 22 14:52 LICENSE
-r--r----- 1 wlynch primarygroup  2484 Nov 22 14:52 README.md
```

# Solution

We should run the equivalent of `chmod +w $(find third_party -type d)` when copying over source in order to make sure directories can be deleted properly on multiple runs of the tool.

### PR DESCRIPTION
**Pull request by dprotaso**
- bump copy dep to latest
- update permissions to fix #11

### CODE REVIEW

File: save.go [hunk 1]
```diff
@@ -118,7 +118,12 @@ func saveMain(_ *cobra.Command, args []string) error {
 func copySrc(src, dest string) error {
 	// Skip the .git directory for copying, if it exists, since we don't want to save the user's
 	// local Git config along with the source code.
-	opt := copy.Options{Skip: func(src string) (bool, error) { return strings.HasSuffix(src, ".git"), nil }}
+	opt := copy.Options{
+		Skip: func(src string) (bool, error) {
+			return strings.HasSuffix(src, ".git"), nil
+		},
+		AddPermission: 0700,
```
Comment (wlynch) [inline save.go:125]: I would imagine we don't need executable permissions on any of these files. Let's use 0600 instead?

**Review by wlynch — COMMENTED** [review]
LGTM! One minor change.

File: save.go (hunk 1 above)
Comment (dprotaso) [reply save.go:125]: cool done

**Review by wlynch — APPROVED** [review]

**Comment by antoineco** [comment]
This PR was merged but the permissions are `0700`, despite the comment that requested a change to `0600`. All copied files are now executable.

**Comment by wlynch** [comment]
> This PR was merged but the permissions are `0700`, despite the comment that requested a change to `0600`. All copied files are now executable.

🤦  

Fixed in https://github.com/google/go-licenses/pull/68

**Comment by dprotaso** [comment]
oh weird - my bad

**Pull request merged**